corehq/apps/userreports/README.md

```diff
@@ -401,6 +413,121 @@ The `date_expression` can be any valid expression, or simply constant
 }
 ```
 
+
+#### Filter, Sort, Map and Reduce Expressions
+
+We have follwoing expressions that act on a list of objects or list of lists. The list to operate on is specified by `items_expression`. This can be any valid expression that returns a list. If the `items_expression` doesn't return a valid list, these might either fail or return one of empty list or `None` value.
+
+##### map_items Expression
+
+`map_items` performs a calculation specified by `map_expression` on each item of the list specified by `items_expression` and returns a list of the calculation results. The `map_expression` is evaluated relative to each item in the list and not relative to the parent document from which the list is specified. For e.g. if `items_expression` is a path to repeat-list of children in a form document, `map_expression` is a path relative to the repeat item.
+
+`items_expression` can be any valid expression that returns a list.
+`map_expression` can be any valid expression relative to the items in above list.
+
+```json
+{
+    "type": "filter_items",
+    "items_expression": {
+        "type": "property_path",
+        "property_path": ["form", "child_repeat"]
+    },
+    "map_expression": {
+        "type": "property_path",
+        "property_path": ["age"]
+    },
+}
+```
+Above returns list of ages. Note that the `property_path` in `map_expression` is realtive to the repeat item rather than to the form.
+
+
+##### filter_items Expression
+
+`filter_items` performs filtering on given list and returns a new list. If the boolean expression specified by `filter_expression` evaluates to a `True` value, the item is included in the new list and if not, is not included in the new list.
+
+`items_expression` can be any valid expression that returns a list.
+`filter_expression` can be any valid boolean expression relative to the items in above list.
+
+```json
+{
+    "type": "filter_items",
+    "items_expression": {
+        "type": "property_name",
+        "property_name": "family_repeat"
+    },
+    "filter_expression": {
+       "type": "boolean_expression",
+        "expression": {
+            "type": "property_name",
+            "property_name": "gender"
+        },
+        "operator": "eq",
+        "property_value": "female"
+    }
+}
+```
+
+##### sort_items Expression
+
+`sort_items` returns a sorted list of items based on sort value of each item in ascending order. The sort value of an item is speicifed by `sort_expression`. If a sort-value of an item is `None`, the item will appear in the start of list. If sort-values of any two items can't be compared, an empty list is returned.
```

Comment:
I don't follow the `"not"` part. My thought process was - currently reduce is the only way to get something out from a list. And it supports both `last_item` and `first_item`. So one could get both max or min without needing to change the order.

Also pretty easy_win to add sort-order latter if needed